Source/WebCore/dom/TrustedTypePolicy.cpp

```diff
@@ -83,15 +81,16 @@ ExceptionOr<Ref<TrustedScriptURL>> TrustedTypePolicy::createScriptURL(const Stri
 ExceptionOr<String> TrustedTypePolicy::getPolicyValue(TrustedType trustedTypeName, const String& input, FixedVector<JSC::Strong<JSC::Unknown>>&& arguments, IfMissing ifMissing)
 {
     CallbackResult<String> policyValue(CallbackResultType::UnableToExecute);
+    Locker locker { lock() };
```

Comment:
I think the scope of this lock is too large which could cause stalls in GC threads. I'd suggest grabbing a RefPtr to the appropriate handler then releasing the lock and calling `handleEvent` on the protected handler.  Something like this:

```
if (trustedTypeName == TrustedType::TrustedHTML) {
    RefPtr<...> handler = nullptr;
    {
        Locker locker { lock() };
        handler = m_options.createHTML;
    }
    if (handler)
        handler->handleEvent(input, WTFMove(arguments));
}
```

You'd probably want to make a helper function for this so you don't have to repeat this pattern for each of the trusted type callbacks.

Reply:
This approach lets us keep the handler alive for the rest of the call via the RefPtr and it allows us to release the lock as soon as we possibly can so the GC threads don't have to wait on the main thread.

Reply:
> You'd probably want to make a helper function for this so you don't have to repeat this pattern for each of the trusted type callbacks.

For now I won't because the callback type is different and so is the property name so a function helper won't be too useful. However, I've raised a spec issue because I don't think we actually need 3 separate types for the callbacks in which case I probably can clean this up a bit.